server-spi-private/src/main/java/org/keycloak/broker/provider/IdentityProviderMapperSyncModeDelegate.java

```diff
@@ -0,0 +1,43 @@
+package org.keycloak.broker.provider;
+
+import org.jboss.logging.Logger;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.IdentityProviderMapperSyncMode;
+import org.keycloak.models.IdentityProviderSyncMode;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+
+public final class IdentityProviderMapperSyncModeDelegate {
+
+    protected static final Logger logger = Logger.getLogger(IdentityProviderMapperSyncModeDelegate.class);
+
+    public static void delegateUpdateBrokeredUser(KeycloakSession session, RealmModel realm, UserModel user, IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context, IdentityProviderMapper mapper) {
+        IdentityProviderSyncMode effectiveSyncMode = combineIdpAndMapperSyncMode(context.getIdpConfig().getSyncMode(), mapperModel.getSyncMode());
+
+        if (!mapper.supportedSyncModes().contains(effectiveSyncMode)) {
+            logger.warnf("The mapper %s does not explicitly support sync mode %s. Please ensure that the SPI supports the sync mode correctly and update it to reflect this.", mapper.getDisplayType(), effectiveSyncMode);
+        }
+
+        if (effectiveSyncMode == IdentityProviderSyncMode.LEGACY) {
+            mapper.updateBrokeredUserLegacy(session, realm, user, mapperModel, context);
+        } else if (effectiveSyncMode == IdentityProviderSyncMode.FORCE) {
+            mapper.updateBrokeredUser(session, realm, user, mapperModel, context);
+        }
+    }
+
+    public static IdentityProviderSyncMode combineIdpAndMapperSyncMode(IdentityProviderSyncMode syncMode, IdentityProviderMapperSyncMode mapperSyncMode) {
+        switch (mapperSyncMode) {
```

Comment:
This can be changed to a single line:
`return IdentityProviderSyncMode.INHERIT.equals(mapperSyncMode) ? syncMode : mapperSyncMode `